notebooks/21_reinforcement_learning/index.md

```diff
@@ -0,0 +1,522 @@
+        utils.EzPickle.__init__(self)
+
+    def step(self, action):
+        xposbefore = self.sim.data.qpos[0]
+        self.do_simulation(action, self.frame_skip)
+        xposafter = self.sim.data.qpos[0]
+        ob = self._get_obs()
+        reward_ctrl = -0.1 * np.square(action).sum()
+        reward_run = (xposafter - xposbefore) / self.dt
+        reward = reward_ctrl + reward_run
+        done = False
+        return ob, reward, done, dict(reward_run=reward_run, reward_ctrl=reward_ctrl)
+
+    def _get_obs(self):
+        return np.concatenate(
+            [
+                self.sim.data.qpos.flat[1:],
+                self.sim.data.qvel.flat,
+            ]
+        )
+
+    def reset_model(self):
+        qpos = self.init_qpos + self.np_random.uniform(
+            low=-0.1, high=0.1, size=self.model.nq
+        )
+        qvel = self.init_qvel + self.np_random.standard_normal(self.model.nv) * 0.1
+        self.set_state(qpos, qvel)
+        return self._get_obs()
+
+    def viewer_setup(self):
+        self.viewer.cam.distance = self.model.stat.extent * 0.5
+```
+The video of this agent in the environment is shown below.
+
+https://user-images.githubusercontent.com/62206142/148566163-089a2a3c-4361-4997-b956-d4e790cf1cbd.mp4
+
+<div id='ProblemwithTD'/>
+
+## Problem with TD
+All we want is to find the best policy that suits us. Although TD agent finds the value of each state (A value that converges to the real value during the time), it cannot find the best policy because for finding that and doing one-step expectimax, $T$ and $R$ functions are needed. However, in RL, they are not available. Therefore, a new method is needed which is called **Q-Learning**.
+
+<div id='Conclusion'/>
+
+# Conclusion
+
+<div id='SummaryofDiscussedRLMethods'/>
+
+### Summary of Discussed RL Methods
+#### Direct Utility Estimation
+* Simple to implement
+* Each update is fast
+* Does not exploit Bellman constraints and converges slowly
+
+#### Temporal Difference Learning
+* Update speed and implementation similar to direct estimation
+* Partially exploits Bellman constraints-adjusts state to 'agree' with the observed successor (Not **all** possible successors)
+
+### Why to use RL
+Here are prime reasons for using Reinforcement Learning:
+
+* It helps you to find which situation needs an action
+* It Helps you to discover which action yields the highest reward over the longer period.
+* It provides the learning agent with a reward function.
+* It allows it to figure out the best method for obtaining large rewards.
+
+### Usage of RL
+Here are applications of Reinforcement Learning:
+
+* Robotics for industrial automation.
+* Business strategy planning
+* Machine learning and data processing
+* It helps you to create training systems that provide custom instruction and materials according to the requirement of students.
+* Aircraft control and robot motion control
+
+<div id='resources'/>
+

```

Comment:
you can have an "Other useful links" section and provide some interesting material related to these topics (just a suggestion)